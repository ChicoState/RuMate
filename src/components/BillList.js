import React, { useState, useEffect } from 'react';
import { View, Text, StyleSheet, FlatList } from 'react-native';
import firebase from 'firebase'
import Bill from '../components/Bill';

const getBills = async (setBills) => {
  const bills = await firebase.database().ref('bills/');
  bills.on('value', (snapshot) => {
    const data = snapshot.val();
    let list = [];
    for (let item in data) {
      if (firebase.auth().currentUser.uid == data[item].uid) {
        list.push(data[item])
      }
    }
    setBills(list);
  })
}

const renderBill = (bills, item) => {
  for (let bill in bills) {
    <Text>
      <Bill 
        name = {bills[bill].name} 
        value = {bills[bill].value}  
      />
    </Text>
  }
}

const renderBills = (bills, setBills) => {
  // firebase pull
  useEffect(() => {
    getBills(setBills);
  }, []);
  console.log(bills);
  <FlatList
    keyExtractor={item => item.uid}
    data={bills}
    renderItem={({item}) => {
      return (
        renderBill(bills, item)
      );
    }}
  />
}

const BillList = () => {
  const [bills, setBills] = useState([]);
  return (
    <View>
      {renderBills(bills, setBills)}
    </View>
  );
}

styles = StyleSheet.create({
  plusButton: {
    fontSize: 20
  }
});

export default BillList;