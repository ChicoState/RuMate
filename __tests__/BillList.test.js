import React from 'react';
import renderer from 'react-test-renderer';

import BillList from '../src/components/BillList';

describe('<BillList />', () => {

  it('has 1 child', () => {
    const tree = renderer.create(<BillList />).toJSON();
    console.log(tree.children.length)
    expect(tree.children.length).toBe(1);
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<BillList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

