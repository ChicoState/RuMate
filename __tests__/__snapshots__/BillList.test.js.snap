// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BillList /> renders correctly 1`] = `
<View>
  <View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 40,
          "paddingTop": "30%",
        }
      }
    >
      No current bills
    </Text>
  </View>
</View>
`;
